src/java/com/unifina/feed/FieldDetector.java

```diff
@@ -37,33 +58,59 @@ public void setFlattenMap(boolean flattenMap) {
 			map = new HashMap();
 		}
 
-		if (flattenMap) {
+		if (flatten) {
 			map = MapTraversal.flatten(map);
 		}
 
-		List<Map<String, String>> fields = new ArrayList<>();
+		List<FieldConfig> fields = new ArrayList<>();
 		for (Object key : map.keySet()) {
```

Comment:
It is faster to iterate thought entrySet() than keySet(). This avoids map lookup overhead.
```
		for (Map.Entry<String, Object> entry : map.entrySet()) {
			fields.add(new FieldConfig(entry.getKey(), detectType(entry.getValue())));
```
Not that it makes a huge difference here.